pilot/pkg/networking/core/v1alpha3/envoyfilter/monitoring.go

```diff
@@ -0,0 +1,49 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package envoyfilter
+
+import (
+	"istio.io/pkg/monitoring"
+)
+
+var (
+	typeTag = monitoring.MustCreateLabel("type")
+
+	totalEnvoyFiltersSkipped = monitoring.NewSum(
+		"pilot_total_envoy_filters_skipped",
+		"Total number of Envoy filters skipped for each proxy.",
```

Comment:
Can we be more clear on what does it mean a EnvoyFilter is skipped? If I understand the change correctly, it seems like the the meaning of skipped is not consistent across types. For example, for listener and filter chain, looks to me it is per resource, but for cluster, it is based on whether the patch is applied on the last cluster of all inbound/outbound clusters.

Also please consider merging skipped / error into one metric, make skipped and error an label, and add an third dimension applied so that we can have full coverage on patching stats.

Reply:
The skipped concept is same. If a filter patch is failed to apply it will be incremented. Please see my other comment.

Moved it to single stat. PTAL